packages/generate-pdf/test/generate-pdf.spec.ts

```diff
@@ -305,7 +271,42 @@ describe('GeneratePDF', function () {
     expect(content[0]).toHaveProperty('layout');
   }, 8000);
 
-  for (const testSuite of testMaps) {
-    runPDFGenerationTests(testSuite);
-  }
+  certificateTestMap.forEach((schema) => {
+    const { type, versions } = schema;
+    const styles = require(`../../generate-${type.toLowerCase()}-pdf-template/utils/styles.js`);
+
+    versions.forEach((version, index) => {
+      const dirPath = resolve(`${__dirname}/../../../fixtures/${type}/${version}`);
+      const files = readdirSync(dirPath);
+      const validCertNames = files.filter((file) => file.match(/^valid_cert_[\d]+.json|^valid_cert.json/));
+      const isLatestVersion = index === versions.length - 1;
+
+      validCertNames.map((validCertName) => {
```

Comment:
The code gets severely nested over here!
Maybe you can create an array of option objects and use that array to run the tests ?
Also we usually use `array.map` when we return something in the callback otherwise you might as well stick with `forEach` or simple `for` loops